examples/pytorch/gat/main.py

```diff
@@ -92,58 +52,81 @@ def train_func(config: Dict):
         optimizer, named_parameters=model.named_parameters()
     )
 
-    address = "localhost:9999"
-    s = Server(address, args.data_dir, 0, len(args.partitions))
-    g = DistributedClient([address])
-    max_id = g.node_count(args.node_type) if args.max_id in [-1, None] else args.max_id
-    dataset = ray.data.range(max_id).repartition(max_id // args.batch_size)
-    pipe = dataset.window(blocks_per_window=4).repeat(args.num_epochs)
+    g = config["get_graph"]()
+    batch_size = 140
+    dataset = ray.data.read_text(f"{config['data_dir']}/train.nodes")
+    dataset = dataset.repartition(dataset.count() // batch_size)
+    pipe = dataset.window(blocks_per_window=4).repeat(config["num_epochs"])
 
     def transform_batch(idx: list) -> dict:
         return model.q.query_training(g, np.array(idx))
 
     pipe = pipe.map_batches(transform_batch)
 
-    for epoch in range(epochs_trained, args.num_epochs):
-        scores = []
-        labels = []
+    test_dataset = ray.data.read_text(f"{config['data_dir']}/test.nodes")
+    test_dataset = test_dataset.repartition(1)
+    test_dataset = test_dataset.map_batches(transform_batch)
+    test_dataset_iter = test_dataset.repeat(config["num_epochs"]).iter_epochs()
+
+    for epoch, epoch_pipe in enumerate(pipe.iter_epochs()):
+        model.train()
+        train_scores = []
```

Comment:
I don't think it is train_scores/labels are very useful for a small dataset you are running this example on. It might be usefull for graphs with millions of nodes, but thousands should be train very fast and only per epoch metrics are valuable.